### ISSUE TITLE
Why does 'Automatic Device Provisioning' not include the device defaults property

### ISSUE THREAD

**Issue by xinbusiqq** [eclipse-hono/hono#3596]
AbstractAutoProvisioningEventSender.sendAutoProvisioningEvent.sendEvent
 set new RegistrationAssertion(deviceId), Without using the Auto-Provisioning device object.

KafkaBasedEventSender.sendEvent.addDefaults, The RegistrationAssertion device.getDefaults() is null.

**Comment by sophokles73** [comment]
It is hard to tell what the issue is that you are facing. Can you elaborate and/or provide a code snippet that illustrates the problem?

**Comment by xinbusiqq** [comment]
ok, 
I added additional properties to the device in 'CA' and registered the device using 'Auto-Provisioning'. These properties are set in the device's defaults attribute, and I want to retrieve the device defaults attribute in 'hono.{tenant}.event'. However, 'AbstractAutoProvisioningEventSender.sendAutoProvisioningEvent.eventSender.sendEvent' does not pass the registered device object; instead, it uses 'new RegistrationAssertion(deviceId)'. As a result, 'KafkaBasedEventSender.sendEvent.addDefaults' cannot access the defaults attribute of the registered device.

**Comment by xinbusiqq** [comment]
> It is hard to tell what the issue is that you are facing. Can you elaborate and/or provide a code snippet that illustrates the problem?

ok,
I added additional properties to the device in 'CA' and registered the device using 'Auto-Provisioning'. These properties are set in the device's defaults attribute, and I want to retrieve the device defaults attribute in 'hono.{tenant}.event'. However, 'AbstractAutoProvisioningEventSender.sendAutoProvisioningEvent.eventSender.sendEvent' does not pass the registered device object; instead, it uses 'new RegistrationAssertion(deviceId)'. As a result, 'KafkaBasedEventSender.sendEvent.addDefaults' cannot access the defaults attribute of the registered device.

**Comment by sophokles73** [comment]
Are you talking about (auto-)registering a new edge device B which connects via a gateway device A that is already registered in the system?
Or are you talking about a new device A which should be auto-registered based on the tenant's trust anchor definition?

**Comment by xinbusiqq** [comment]
> Are you talking about (auto-)registering a new edge device B which connects via a gateway device A that is already registered in the system? Or are you talking about a new device A which should be auto-registered based on the tenant's trust anchor definition?


Yes, I'm referring to the auto-registration based on the tenant's trust anchor definition.
The following is a notification emitted by the 'hono.event.testTenant' topic after successful registration. The notification includes the 'types' property of the 'defaults' attribute for 'testTenant', but does not include the 'defaults' attribute for the new device.
```json
header: 
	[ key 1:  tenant_id, value 1: datang ]
	[ key 2:  types, value 2: {"obu":{}} ] 
	[ key 3:  orig_adapter, value 3: hono-device-registry ] 
	[ key 4:  device_id, value 4: test:obu002 ]
	[ key 5:  qos, value 5: 1 ]
	[ key 6:  hono_registration_status, value 6: NEW ]
	[ key 7:  content-type, value 7: application/vnd.eclipse-hono-device-provisioning-notification ]
	[ key 8:  orig_address, value 8: event ]
	[ key 9:  creation-time, value 9: 1702609334479 ]
	[ key 10:  traceparent, value 10: 00-13afddcfc23ff877edf3e18393621517-605de170bef2cfce-00 ]
```
Because the 'AbstractAutoProvisioningEventSender.sendAutoProvisioningEvent' method does not take the object of the new device as a parameter, and the 'eventSender.sendEvent()' method uses 'new RegistrationAssertion(deviceId)' as second parameter, so the 'defaults' property of the registrated device is not included.

**Comment by sophokles73** [comment]
Ok, you have defined the `types` default property for the tenant that the device is being auto-provisioned to, right? And then device `test:obu002` is being auto-provisioned to this tenant when it first connects, right?

>  but does not include the 'defaults' attribute for the new device.
Where should these default properties come from? There is no device information available yet in the registry, that's the whole point of doing auto-registration, isn't it?

**Comment by xinbusiqq** [comment]
I have expanded the auto-provisioned functionality, adding some default information to the certificate for the device. I believe it is better to use the object of device registration rather than 'new RegistrationAssertion(deviceId),' as I want to obtain more information from the automatic registration notification events.

**Comment by sophokles73** [comment]
Ok, so what is this issue all about then?
